core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -502,6 +502,9 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     _applicationId = _taskScheduler.applicationId()
     _applicationAttemptId = taskScheduler.applicationAttemptId()
     _conf.set("spark.app.id", _applicationId)
+    if (_conf.getBoolean("spark.ui.reverseProxy", false)) {
+      System.setProperty("spark.ui.proxyBase", "/target/" + _applicationId)
```

Comment:
I wonder if this is better named /proxy/app/applicationId for applications, and /proxy/worker/workerId, so it's more clear what the destination target is?